src/main/java/datastax/astra/migrate/CopyJobSession.java

```diff
@@ -126,13 +123,24 @@ public void getDataAndInsert(BigInteger min, BigInteger max) {
         }
     }
 
+    public synchronized void printCounts(boolean isFinal) {
+        String msg = "TreadID: " + Thread.currentThread().getId();
+        if (isFinal) {
+            msg += " Final";
+            logger.info("################################################################################################");
+        }
+        logger.info(msg + " Read Record Count: " + readCounter.get());
```

Comment:
#nitpick I'd change this to use substitutions as opposed to concatinations for efficiency like,

```
logger.info("{} Read Record Cont: {}", msg, readCounter.get());
```